protocols/kad/src/kbucket.rs

```diff
@@ -308,10 +273,10 @@ where
         BucketsIter(self.tables.iter_mut(), self.unresponsive_timeout)
     }
 
-    /// Finds the nodes closest to `id`, ordered by distance.
+    /// Finds the keys closest to `key`, ordered by distance.
     ///
     /// Pending nodes are ignored.
-    pub fn find_closest(&mut self, id: &impl KBucketsPeerId<TPeerId>) -> VecIntoIter<TPeerId> {
+    pub fn find_closest<T>(&mut self, key: &Key<T>) -> VecIntoIter<Key<TPeerId>> {
```

Comment:
I was curious what the code of this method would look like now. If I understand correctly, it will now be possible to write an optimized version of this (but it's not part of this PR)?

Reply:
Oh nevermind, I was mixing with #1117

Reply:
Right, that is in https://github.com/libp2p/rust-libp2p/pull/1117 which I'm still polishing up, but the part that deals with this optimisation in particular is stable and [can be found here](https://github.com/libp2p/rust-libp2p/blob/80cc7eaa78a62e291842e149f52c47659260eb42/protocols/kad/src/kbucket.rs#L259-L447). The actual `Iterator` impl for the (generic) `ClosestIter` is at the end of that highlighted code fragment - it just draws from two underlying iterators. The actual ordering of the bucket indices is provided by the (internal) `ClosestBucketsIter`.